### ISSUE TITLE
Crash with RewardsServiceImpl::CreateRewardsWallet in the perf tests (with --enable-brave-features-for-perf-testing)

### ISSUE THREAD

**Issue by atuchin-m** [brave/brave-browser#37252]
A local crash when launching the perf tests. The perf tests uses `--enable-brave-features-for-perf-testing` to enable a set of feature, including rewards.
```
mikhail@am-macbook-2 /tmp % brave-nightly  --enable-brave-features-for-perf-testing --user-data-dir=/tmp/pp1
[93347:259:0402/154311.381377:ERROR:policy_logger.cc(157)] :components/enterprise/browser/controller/chrome_browser_cloud_management_controller.cc(161) Cloud management controller initialization aborted as CBCM is not enabled. Please use the `--enable-chrome-browser-cloud-management` command line flag to enable it if you are not using the official Google Chrome build.
[0402/154311.610617:WARNING:crash_report_exception_handler.cc(235)] UniversalExceptionRaise: (os/kern) failure (5)
zsh: segmentation fault   --enable-brave-features-for-perf-testing --user-data-dir=/tmp/pp1
```
The stack:
```
[ 00 ] std::__Cr::unique_ptr<base::OneShotEvent, std::__Cr::default_delete<base::OneShotEvent>>::operator->() const ( unique_ptr.h:259 )
[ 01 ] brave_rewards::RewardsServiceImpl::CreateRewardsWallet(std::__Cr::basic_string<char, std::__Cr::char_traits<char>, std::__Cr::allocator<char>> const&, base::OnceCallback<void (brave_rewards::mojom::CreateRewardsWalletResult)>) ( rewards_service_impl.cc:564 )
[ 02 ] perf::MaybeEnableBraveFeatureForPerfTesting(Profile*) ( brave_perf_features_processor.cc:69 )
[ 03 ] BraveProfileManager::InitProfileUserPrefs(Profile*) ( brave_profile_manager.cc:98 )
[ 04 ] ProfileImpl::OnLocaleReady(Profile::CreateMode) ( profile_impl.cc:1144 )
[ 05 ] ProfileImpl::OnPrefsLoaded(Profile::CreateMode, bool) ( profile_impl.cc:1219 )
[ 06 ] ProfileImpl::ProfileImpl(base::FilePath const&, Profile::Delegate*, Profile::CreateMode, base::Time, scoped_refptr<base::SequencedTaskRunner>) ( profile_impl.cc:524 )
[ 07 ] ProfileImpl::ProfileImpl(base::FilePath const&, Profile::Delegate*, Profile::CreateMode, base::Time,
```

**Comment by zenparsing** [comment]
@atuchin-m  I enabled the flag for a local build, but have not yet been able to reproduce the crash. I noticed that [here](https://github.com/brave/brave-core/blob/a79fb87f4f3b5b0c833644ed870a657eef74b987/browser/perf/brave_perf_features_processor.cc#L69) we aren't checking if the `RewardsService` instance is a `nullptr`, and we also have no `DCHECK`. Could it be that there is no instance for this profile?
